lib/rules/mustache-curly-spacing.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview Enforce spacing on the style of mustache curly brackets.
+ * @author Armano
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Enforce spacing on the style of mustache curly brackets.',
+      category: 'Stylistic Issues',
+      recommended: false
+    },
+    fixable: 'whitespace',
+    schema: [
+      {
+        enum: ['always', 'never']
+      }
+    ]
+  },
+
+  create (context) {
+    const options = context.options[0]
+    const optSpaces = options !== 'never'
+    const template = context.parserServices.getTemplateBodyTokenStore && context.parserServices.getTemplateBodyTokenStore()
+
+    // ----------------------------------------------------------------------
+    // Helpers
+    // ----------------------------------------------------------------------
+
+    function checkTokens (leftToken, rightToken) {
+      if (leftToken.loc.end.line === rightToken.loc.start.line) {
+        const spaces = rightToken.loc.start.column - leftToken.loc.end.column
+        if (optSpaces === (spaces === 0)) {
+          context.report({
+            node: rightToken,
+            loc: {
+              start: leftToken.loc.end,
+              end: rightToken.loc.start
+            },
+            message: 'Found {{spaces}} whitespaces, {{type}} expected.',
+            data: {
+              spaces: spaces === 0 ? 'none' : spaces,
+              type: optSpaces ? '1' : 'none'
+            },
+            fix: (fixer) => fixer.replaceTextRange([leftToken.range[1], rightToken.range[0]], optSpaces ? ' ' : '')
+          })
+        }
+      }
+    }
+
+    // ----------------------------------------------------------------------
+    // Public
+    // ----------------------------------------------------------------------
+
+    utils.registerTemplateBodyVisitor(context, {
+      'VExpressionContainer[expression!=null]' (node) {
+        const tokens = template.getTokens(node, {
+          includeComments: true,
+          filter: token => token.type !== 'HTMLWhitespace' // When there is only whitespace between ignore it
+        })
+
+        const startToken = tokens.shift()
```

Comment:
Do you intentionally mutate `tokens` here? Using that kind of methods usually introduce more confusion and may lead to unexpected results.